### PR TITLE
Add non-prod zones

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###

[AB#883]

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/demo/local.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+locals {
```
Comment (timja) [inline components/demo/local.tf:1]: this should be in the module not duplicated everywhere

File: components/aat/aat-idam-internal.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module "aat-idam-internal" {
+  source              = "../../modules/azure-private-dns/"
+  cname_records       = yamldecode(data.local_file.aat-idam-internal-configuration.content).cname
+  a_recordsets        = yamldecode(data.local_file.aat-idam-internal-configuration.content).A
+  zone_name           = yamldecode(data.local_file.aat-idam-internal-configuration.content).name
+  vnet_links          = yamldecode(data.local_file.aat-idam-internal-configuration.content).vnet_links
+  resource_group_name = var.resource_group_name
+  env                 = var.env
+  common_tags         = common_tags
```
Comment (timja) [inline components/aat/aat-idam-internal.tf:9]: shouldn't this line be deleted?

Comment (luigibk) [reply components/aat/aat-idam-internal.tf]: Indeed ... and the related var too.

File: components/sandbox/provider.tf [hunk 3]
```diff
@@ -1,8 +1,6 @@
 provider "azurerm" {
   version = "=2.3.0"
   features {}
-}
 
-terraform {
-  backend "azurerm" {}
-}
\ No newline at end of file
+  subscription_id = "1497c3d7-ab6d-4bb7-8a10-b51d03189ee3"
```
Comment (timja) [inline components/sandbox/provider.tf:5]: why is this here?

Comment (luigibk) [reply components/sandbox/provider.tf]: Local file
